_blogposts/2024-02-01-release-11-1-0.mdx

```diff
@@ -0,0 +1,257 @@
+
+var filename = "index.res";
+
+var result = await $$Bun.$`ls ${filename}`;
+```
+
+</CodeTab>
+
+
+Of course you can also create your own tag functions in ReScript now as well, it is just a function with the following signature.
+
+```rescript
+let myTagFunction : (array<string>, array<'param>) => 'output
+```
+
+Refer to the docs to find a [detailed example](/docs/manual/latest/tagged-templates).
+
+## Import attributes
+
+Import attributes is a JS feature that is [currently in standardization](https://github.com/tc39/proposal-import-attributes), but is already implemented by many JS tools. Now, ReScript supports it too, as long as the compiler is configured to output ES6.
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res
+@module({from: "./myJson.json", with: {type_: "json", \"some-identifier": "yep"}})
+external myJson: Js.Json.t = "default"
+Console.log(myJson)
+
+@module({from: "./myCss.css", with: {type_: "css", \"some-identifier": "yep"}})
+external buttonCss: string = "button"
+Console.log(buttonCss)
+```
+```js
+import * as MyCssCss from "./myCss.css" with {"type": "css", "some-identifier": "yep"};
+import MyJsonJson from "./myJson.json" with {"type": "json", "some-identifier": "yep"};
+
+var myJson = MyJsonJson;
+console.log(myJson);
+
+var buttonCss = MyCssCss.button;
+console.log(buttonCss);
+```
+</CodeTab>
+
+## BigInt support
+
+ReScript now natively supports [JavaScript's `bigint` type](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt).
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+open! Js.BigInt 
+
+let a = 9007199254740991n + 9007199254740991n
+let b = 2n ** 2n
+```
+```js
+var a = 9007199254740991n + 9007199254740991n;
+
+var p = 2n ** 2n;
+```
+
+</CodeTab>
+
+See [big integer docs](https://rescript-lang.org/docs/manual/latest/primitive-types#big-integers-experimental) for more.
+
+## Array spread syntax
+
+The spread syntax, which was already supported for records and lists for a long time, now also supports arrays!
+
+```rescript
+let animals = ["🐶", "🐱", "🐷"]
+let moreAnimals = [...animals, "🐔", "🐴", "🐮"]
+```
+
+## Hyphens in JSX tag names
+
+We lifted restrictions on JSX tag names. This means you no longer need to escape tag names that contain hyphens:
+
+Previously:
+
+```rescript
+let x = <\"custom-tag" />
+```
+
+Now:
+
+```rescript
+let x = <custom-tag />
+```
+
+This is particularly useful when dealing with [web components](https://developer.mozilla.org/en-US/docs/Web/API/Web_components), where element names tend to use hyphens.
+
+## Omit trailing undefined in external function calls
+
+ReScript 11's uncurried mode allows for much more ergonomic external function bindings, because trailing units were not needed anymore. But, this comes with a potential problem. All arguments, whether they're actually supplied or not, were printed as `undefined` in the resulting JS. This is handled better now, as trailing `undefined`s are automatically omitted.
```

Comment:
```suggestion
ReScript 11's uncurried mode allows for much more ergonomic external function bindings, because trailing units are not needed anymore. But, this comes with a potential problem. All arguments, whether they're actually supplied or not, were printed as `undefined` in the resulting JS. This is handled better now, as trailing `undefined`s are automatically omitted.
```